### PR TITLE
Add @runtimeCompile attribute into druntime

### PR DESCRIPTION
**Pull request by Hardcode84**

### CODE REVIEW

**Comment by thewilsonator** [comment]
Needs a doc comment, probably with a note saying it is WiP. otherwise looks good.

**Comment by Hardcode84** [comment]
> Needs a doc comment

Yeah I know, I will add it when we decide with naming

**Comment by thewilsonator** [comment]
`runtimeCompile` is a sensible enough name for now. We can always change it later if need be.

**Comment by Hardcode84** [comment]
> Needs a doc comment

Done

**Comment by thewilsonator** [comment]
LGTM.

File: src/ldc/attributes.d [hunk 1]
```diff
@@ -238,3 +238,30 @@ immutable weak = _weak();
 private struct _weak
 {
 }
+
+/++
+ + When applied to a function, marks this function for dynamic compilation.
+ + All functions marked with this attribute must be explicitly compiled in
+ + runtime via ldc.runtimecompile api before usage.
+ +
+ + When applied to global variable, this variable will be treated as constant
```
Comment (dnadlinger) [inline src/ldc/attributes.d:247]: Perhaps it would be clearer to separate out this second meaning into another attribute? `runtimeCompileConstant` maybe?

**Pull request merged**